connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/ResponseValidator.scala

```diff
@@ -30,6 +33,12 @@ class ResponseValidator extends Logging {
   // do not use server-side streaming.
   private var serverSideSessionId: Option[String] = None
 
+  // Indicates whether the client and the client information on the server do not correspond to each
+  // other. This flag being true means that the server has restarted and lost the client
+  // information, or there is a logic error in the code; both cases, the user should establish a new
+  // connection to the server. Access to the value has to be synchronized since it can be shared.
+  private val sessionInvalidated: AtomicBoolean = new AtomicBoolean(false)
```

Comment:
I'm not intimately familiar with this code to tell whether this is the right place to keep session state. However, based on the precedence that "serverSideSessionId" is here, I'm fine with this.

Reply:
My minor suggestion would be to invert this boolean and call it "isSessionActive" along the lines of Scala boolean naming conventions.

Reply:
Thanks, that will be nicer.